packages/core/core-flows/src/order/workflows/order-edit/update-order-edit-item-quantity.ts

```diff
@@ -0,0 +1,118 @@
+  transform,
+} from "@medusajs/workflows-sdk"
+import { useRemoteQueryStep } from "../../../common"
+import {
+  previewOrderChangeStep,
+  updateOrderChangeActionsStep,
+} from "../../steps"
+import {
+  throwIfIsCancelled,
+  throwIfOrderChangeIsNotActive,
+} from "../../utils/order-validation"
+
+/**
+ * This step validates that an item can be updated from an order edit.
+ */
+export const updateOrderEditItemQuantityValidationStep = createStep(
+  "update-order-edit-update-quantity-validation",
+  async function (
+    {
+      order,
+      orderChange,
+      input,
+    }: {
+      order: OrderDTO
+      orderChange: OrderChangeDTO
+      input: OrderWorkflow.UpdateOrderEditItemQuantityWorkflowInput
+    },
+    context
+  ) {
+    throwIfIsCancelled(order, "Order")
+    throwIfOrderChangeIsNotActive({ orderChange })
+
+    const associatedAction = (orderChange.actions ?? []).find(
+      (a) => a.id === input.action_id
+    ) as OrderChangeActionDTO
+
+    if (!associatedAction) {
+      throw new Error(
+        `No request to update item quantity for order ${input.order_id} in order change ${orderChange.id}`
+      )
+    } else if (associatedAction.action !== ChangeActionType.ITEM_UPDATE) {
+      throw new Error(`Action ${associatedAction.id} is not updating an item`)
+    }
+  }
+)
+
+export const updateOrderEditItemQuantityWorkflowId =
+  "update-order-edit-update-quantity"
+/**
+ * This workflow updates a new item in the order edit.
+ */
+export const updateOrderEditItemQuantityWorkflow = createWorkflow(
+  updateOrderEditItemQuantityWorkflowId,
+  function (
+    input: WorkflowData<OrderWorkflow.UpdateOrderEditItemQuantityWorkflowInput>
+  ): WorkflowResponse<OrderPreviewDTO> {
+    const order: OrderDTO = useRemoteQueryStep({
+      entry_point: "orders",
+      fields: ["id", "status", "canceled_at", "items.*"],
+      variables: { id: input.order_id },
+      list: false,
+      throw_if_key_not_found: true,
+    }).config({ name: "order-query" })
+
+    const orderChange: OrderChangeDTO = useRemoteQueryStep({
+      entry_point: "order_change",
+      fields: ["id", "status", "version", "actions.*"],
+      variables: {
+        filters: {
+          order_id: input.order_id,
+          status: [OrderChangeStatus.PENDING, OrderChangeStatus.REQUESTED],
+        },
+      },
+      list: false,
+    }).config({ name: "order-change-query" })
+
+    updateOrderEditItemQuantityValidationStep({
+      order,
+      input,
+      orderChange,
+    })
+
+    const updateData = transform(
+      { orderChange, input },
+      ({ input, orderChange }) => {
+        const originalAction = (orderChange.actions ?? []).find(
+          (a) => a.id === input.action_id
+        ) as OrderChangeActionDTO
+
+        const data = input.data
+        return {
+          id: input.action_id,
+          details: {
+            quantity: data.quantity ?? originalAction.details?.quantity,
+          },
+          internal_note: data.internal_note,
```

Comment:
q: should we add reason here?

Reply:
We currently don't have a reason to associate the order edit. We only have return reason and claim reason.
What do you have in mind?

Reply:
Ah I see, the reason would be added directly to the order. Makes sense